trap/src/main/java/com/cursorinsight/trap/NetworkTrapConfigProvider.kt

```diff
@@ -1,4 +1,33 @@
 package com.cursorinsight.trap
 
-class NetworkTrapConfigProvider {
+import android.app.Application
+import android.util.Log
+import com.google.gson.Gson
+import java.net.HttpURLConnection
+import java.net.URL
+
+open class NetworkTrapConfigProvider(
+    private var configUrl: URL
+) : TrapConfigProvider {
+    override fun getConfig(application: Application): TrapConfig {
+        try {
+            val configString = downloadConfig()
+            return Gson().fromJson(configString, TrapConfig::class.java)
+        } catch (e: Exception) {
+            Log.e(
+                TrapApplication::class.simpleName,
+                "Could not initialize the application"
```

Comment:
I recommend: "Could not download or parse configuration"

...or...

add an additional "try .. catch .." block later to catch downloading related issues separately.

Reply:
Done it was a copy paste error